front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx

```diff
@@ -165,144 +73,200 @@ const RollingStockEditorForm = ({
-  const [isCurrentEffortCurveDefault, setIsCurrentEffortCurveDefault] = useState<boolean>(true);
 
-  useEffect(() => {
-    if (rollingStockData) {
-      setCurrentRsEffortCurve(rollingStockData.effort_curves);
-      setIsCurrentEffortCurveDefault(false);
-    }
-  }, [rollingStockData]);
-
-  // here we generate the resolver for the form validation
-  const resolver = ajvResolver(schema, {
-    formats: fullFormats,
-    coerceTypes: true,
-  });
-
-  // here we generate the hook to set the entire form with a register for each field
-  const {
-    register,
-    handleSubmit,
-    formState: { errors },
-  } = useForm<RollingStockParametersValues>({
-    resolver: resolver as Resolver<RollingStockParametersValues, unknown>,
-    defaultValues: {
-      name: rollingStockData?.name || '',
-      railjson_version: rollingStockData?.railjson_version || '',
-      detail: rollingStockData?.metadata.detail || '',
-      family: rollingStockData?.metadata.family || '',
-      grouping: rollingStockData?.metadata.grouping || '',
-      number: rollingStockData?.metadata.number || '',
-      reference: rollingStockData?.metadata.reference || '',
-      series: rollingStockData?.metadata.series || '',
-      subseries: rollingStockData?.metadata.subseries || '',
-      type: rollingStockData?.metadata.type || '',
-      unit: rollingStockData?.metadata.unit || '',
-      length: rollingStockData?.length || 0,
-      mass: rollingStockData ? rollingStockData.mass / 1000 : 0, // The mass received is in kg and should appear in tons.
-      max_speed: rollingStockData ? rollingStockData.max_speed * 3.6 : 0, // The speed received is in m/s and should appear in km/h.
-      startup_time: rollingStockData?.startup_time || 0,
-      startup_acceleration: rollingStockData?.startup_acceleration || 0,
-      comfort_acceleration: rollingStockData?.comfort_acceleration || 0,
-      gamma_value: rollingStockData?.gamma.value || 0,
-      inertia_coefficient: rollingStockData?.inertia_coefficient || 0,
-      loading_gauge: (rollingStockData?.loading_gauge as string) || 'G1',
-      rolling_resistance_A: rollingStockData?.rolling_resistance.A || 0,
-      rolling_resistance_B: rollingStockData?.rolling_resistance.B || 0,
-      rolling_resistance_C: rollingStockData?.rolling_resistance.C || 0,
-      electrical_power_startup_time: rollingStockData?.electrical_power_startup_time || null,
-      raise_pantograph_time: rollingStockData?.raise_pantograph_time || null,
-      default_mode: rollingStockData?.effort_curves.default_mode || selectedMode,
-      effort_curves: {
-        modes: {
-          [`${selectedMode}`]: {
-            curves: defaultRollingstockMode.modes[`${selectedMode}`].curves,
-            is_electric: defaultRollingstockMode.modes[`${selectedMode}`].is_electric,
-            default_curve: defaultRollingstockMode.modes[`${selectedMode}`].default_curve,
-          },
+  const defaultValues: RollingStockParametersValues = {
```

Comment:
Can you use a useMemo here to avoid instantiating this const at each re-render ? Also, to reduce the number of lines of this file, maybe it will be useful to create a file utils with a default rollingStock.

```suggestion
  // in utils
  export const rollingStockEditorDefaultValues = {
    railjsonVersion: '',
    name: '',
    ...
  };
  
  // in RollingStockEditorForm
  const defaultValues: RollingStockParametersValues = useMemo(() => {
    if (rollingStockData) {
      const modes = { [`${selectedMode}`]: {
          curves: defaultRollingstockMode.modes[`${selectedMode}`].curves,
          isElectric: defaultRollingstockMode.modes[`${selectedMode}`].is_electric,
          defaultCurve: defaultRollingstockMode.modes[`${selectedMode}`].default_curve,
        } };
        return { ...rollingStockData, effortCurves: { modes } };
    }
    return rollingStockEditorDefaultValues;
  }, [rollingStockData, selectedMode, defaultRollingStockMode])
```